index.js

```diff
@@ -288,38 +321,40 @@ async function run() {
     const taskDefArn = registerResponse.taskDefinition.taskDefinitionArn;
     core.setOutput('task-definition-arn', taskDefArn);
 
-    // Update the service with the new task definition
-    if (service) {
-      const clusterName = cluster ? cluster : 'default';
+    const clusterName = cluster ? cluster : 'default';
 
-      // Determine the deployment controller
-      const describeResponse = await ecs.describeServices({
-        services: [service],
-        cluster: clusterName
-      }).promise();
+    if (!serviceInput) {
+      core.debug('Service was not specified, no service updated');
+      return;
+    }
 
-      if (describeResponse.failures && describeResponse.failures.length > 0) {
-        const failure = describeResponse.failures[0];
-        throw new Error(`${failure.arn} is ${failure.reason}`);
-      }
+    const { services, failures } = await ecs.describeServices({cluster, services: [serviceInput]}).promise();
+    const [service] = services;
+    const [failure] = failures;
 
-      const serviceResponse = describeResponse.services[0];
-      if (serviceResponse.status != 'ACTIVE') {
-        throw new Error(`Service is ${serviceResponse.status}`);
-      }
+    if (!serviceConfig) {
+      const createInfo = "If you want the service to be created, enable the 'create-service-if-not-found' input field.";
+      if (failure) throw new Error(`${failure.arn} is ${failure.reason}. ${failure.reason === 'MISSING' ? createInfo : ''}`);
 
-      if (!serviceResponse.deploymentController || !serviceResponse.deploymentController.type || serviceResponse.deploymentController.type === 'ECS') {
-        // Service uses the 'ECS' deployment controller, so we can call UpdateService
-        await updateEcsService(ecs, clusterName, service, taskDefArn, waitForService, waitForMinutes, forceNewDeployment);
-      } else if (serviceResponse.deploymentController.type === 'CODE_DEPLOY') {
-        // Service uses CodeDeploy, so we should start a CodeDeploy deployment
-        await createCodeDeployDeployment(codedeploy, clusterName, service, taskDefArn, waitForService, waitForMinutes);
-      } else {
-        throw new Error(`Unsupported deployment controller: ${serviceResponse.deploymentController.type}`);
-      }
-    } else {
-      core.debug('Service was not specified, no service updated');
+      if (service.status != 'ACTIVE') throw new Error(`Service is ${service.status}`);
+    }
+
+    const config = service || serviceConfig;
+
+    validateDeploymentController(config);
+
+    if (!service) {
```

Comment:
Eu entendo que se caso o createServiceIfNotFound fosse false, o serviceInput provavelmente não seria passado e ele não chegaria nessa parte, parando na linha 327. Porém pode acontecer de que os inputs não sejam tão bem compreendidos e quem for chamar a action passe o nome do service que deseja atualizar porém não tenha a intenção de criar o service caso ele não exista (passou o nome errado por exemplo), dessa forma iria criar se não incluir essa variável no if.

Fiz o teste com essas configurações e o service foi criado (pois não existia o 3 ainda)
![image](https://github.com/eteg/ecs-deploy-task-definition/assets/68859964/c62bfb00-5e30-4263-8389-7e321442bd05)


```suggestion
    if (!service && createServiceIfNotFound) {
```

Reply:
Hummm, nesse caso eu acredito que seria melhor eu corrigir a lógica dos inputs, não?

Pois o único intuito do service-config-file, seria a config que o NOVO service deve possuir

Essa action não deve ser responsável por atualizar configurações de service existentes, pois a action original já não é responsável por este tipo de ação... Por isso não faria muito sentido passar essa config ai
Ela até trabalha com atualizações no service, mas somente de atribuir a nova task definition

Permitir o service-config-file sem possuir o create-service-if-not-found ativo, seria uma falha na lógica dos inputs realmente...

Reply:
Acho que faz sentido. Estou de acordo